src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -863,6 +925,74 @@ private void TextBoxBase_OnTextChanged(object sender, TextChangedEventArgs e)
             LogScroller.ScrollToBottom();
         }
 
+        private void LoadStateMenus(object sender, RoutedEventArgs e)
+        {
+            //grab serialized designoptions from current workspace... hopefully this is loaded?
+            var PresetSet = dynamoViewModel.Model.CurrentWorkspace.PresetsCollection;
+            // now grab all the states off the set and create a menu item for each one
+
+           
+            //only update the states menus if the states have been updated or the user
+            // has switched workspace contexts, can check if designOptionsSet collection is different
+            if (!PresetSet.PresetStates.SequenceEqual(((MenuItem)(sender)).Items.OfType<MenuItem>().Select(x=>x.Tag).ToList()))
```

Comment:
This LINQ statement is very hard to read, an likely hard to debug.  

Instead of using inline c-style casts, use `OfType<T>`